python/tests/unit/arcticdb/version_store/test_append.py

```diff
@@ -280,3 +282,173 @@ def test_(initial: InputFactories, append: InputFactories, match, swap, lmdb_ver
+        merged_segment_row_size,
+        df_in_str,
+        cycle,
+        before_compact,
+    ):
+        index_offset = cycle * segment_row_size * segment_count
+        for count in range(0, segment_count):
+            df = get_wide_and_long_df(
+                index_offset + count * segment_row_size,
+                index_offset + (count + 1) * segment_row_size,
+                col_per_append_df,
+                col_name_set,
+                df_in_str,
+            )
+            before_compact = pd.concat([before_compact, df])
+            if count == 0 and cycle == 0:
+                lib.write(sym, df)
+            else:
+                lib.append(sym, df)
+            segment_details = lib.read_index(sym)
+            assert lib.is_symbol_fragmented(sym, None) is (
+                get_no_of_segments_after_defragmentation(segment_details, merged_segment_row_size)
+                != get_no_of_column_merged_segments(segment_details)
+            )
+        assert lib.is_symbol_fragmented(sym, None) is True
+        seg_details_before_compaction = lib.read_index(sym)
+        lib.defragment_symbol_data(sym, None)
+        res = lib.read(sym).data
+        res = res.reindex(sorted(list(res.columns)), axis=1)
+        res = res.replace("", 0.0)
+        res = res.fillna(0.0)
+        before_compact = before_compact.reindex(sorted(list(before_compact.columns)), axis=1)
+        before_compact = before_compact.fillna(0.0)
+
+        seg_details = lib.read_index(sym)
+
+        assert_frame_equal(before_compact, res)
+
+        assert len(seg_details) == get_no_of_segments_after_defragmentation(
+            seg_details_before_compaction, merged_segment_row_size
+        )
+        indexs = (
+            seg_details["end_index"].astype(str).str.rsplit(" ", n=2).agg(" ".join).reset_index()
+        )  # start_index and end_index got merged into one column
+        assert np.array_equal(indexs.iloc[1:, 0].astype(str).values, indexs.iloc[:-1, 1].astype(str).values)
+        return before_compact
+
+    set_config_int("SymbolDataCompact.SegmentCount", num_of_segments_to_compact_threshold)
+    for dynamic_schema, libs in [
+        (
+            True,
+            [
+                lmdb_version_store_tiny_segment_dynamic,
+                lmdb_version_store_column_buckets,
+                lmdb_version_store_tiny_segment_dynamic_dynamic_string,
+            ],
+        ),
+        (
+            False,
+            [
+                lmdb_version_store_tiny_segment,
+                lmdb_version_store_column_buckets_dynamic_string,
+                lmdb_version_store_tiny_segment_dynamic_string,
+            ],
+        ),
+    ]:
+        for lib in libs:
+            for segment_row_size in [1, 3]:
+                for df_in_str in [True, False]:
+                    for _ in range(num_of_repeat):
+                        before_compact = pd.DataFrame()
+                        for cycle in range(num_of_compact_per_cycle):
```

Comment:
Don't reinvent `parameterize` or `hypothesis`. Please use the `version_store_factory` method I mentioned above.

Reply:
I will revamp this test method per your suggestions.